Consistent processing when the target file is missing.

MarkGibbons opened the pull request:
Add the ignore_missing option
Set the default for ignore_missing to true
Add tests of processing missing files with and without ignore_missing set to false
This change restores the processing of missing files to the 1.0.0 behavior.  We flipped with with 1.1.0 release and have gone back to the old behavior with 2.0.0

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sous-chefs/line-cookbook/99)
<!-- Reviewable:end -->

Commit:
Consistent processing when the target file is missing.

Add the ignore_missing option
Set the default for ignore_missing to true
Add tests of processing missing files with and without ignore_missing set to false
This change restores the processing of files to the 1.0.0 behavior.  We flipped with with 1.1.0 release and have gone
back to the old behavior with 2.0.0

damacus commented on .kitchen.dokken.yml:
```diff
@@ -26,20 +26,25 @@ platforms:
     pid_one_command: /bin/systemd
 
 suites:
```
All of the dokken suites can be removed as we use KITCHEN_LOCAL to override certain sections. We'll just manage suites in the main kitchen.yml

MarkGibbons replied:
Cool, that is a lot of duplicate work.

MarkGibbons replied:
Done.

Commit:
Consistent handling of missing files

damacus commented on resources/delete_from_list.rb:
```diff
@@ -2,15 +2,16 @@
 action :edit do
-  return if !::File.exist?(new_resource.path) && new_resource.ignore_missing
-  raise "File #{new_resource.path} not found" unless ::File.exist?(new_resource.path)
+  file_exist = ::File.exist?(new_resource.path)
```
what do you think about moving file_exist(s) into the helpers and removing that duplication?

MarkGibbons replied:
Sure

damacus commented on spec/unit/recipes/add_to_list_missing_file_spec.rb:
```diff
@@ -20,7 +20,7 @@
 require 'chefspec/berkshelf'
 
-describe 'test::add_to_list_missing_file' do
+describe 'test::add_to_list_missing_file_rspec' do
```
rspec?

MarkGibbons replied:
Working on the names.  Those files get used by the rspec tests.  I'll combine the other files.

damacus replied:
In that case shouldn't they be "inspec"? 

Having the name at the end seems superfluous .

MarkGibbons replied:
I'm going to move them over to the spec directory. They get used by running rspec.
I'll fix the names.

MarkGibbons replied:
fixed

Commit:
Fix the missing file test.

Move the chefspec test recipes to spec/fixtures
Merge the missing file tests with the other tests for each provider

Commit:
Remove the test suites from .kitchen.dokken.yml

Test kitchen will merge the test suites from .kitchen.yml for the travis tests.
